docs/reference/aggregation.md

```diff
@@ -0,0 +1,488 @@
+            "value": 35.97893635571055
+          },
+          "doc_count": 28675,
+          "key": "花蓮縣"
+        }
+      ],
+      "sum_other_doc_count": 1872055
+    }
+```
+#### Limitations
+
+Currently aggregations work only on single value fast fields of type u64, f64, i64 and on string fields.
+
+
+
+### Supported Aggregations
+
+ - Bucket
+    - [Histogram](#histogram)
+    - [Range](#range)
+    - [Terms](#terms)
+- Metric
+    - [Average](#average)
+    - [Stats](#stats)
+
+
+## Bucket Aggregations
+
+BucketAggregations create buckets of documents. Each bucket is associated with a rule which determines whether or not a document in the falls into it. 
+In other words, the buckets effectively define document sets. Buckets are not necessarily disjunct, therefore a document can fall into multiple buckets. 
+In addition to the buckets themselves, the bucket aggregations also compute and return the number of documents for each bucket. 
+Bucket aggregations, as opposed to metric aggregations, can hold sub-aggregations. 
+These sub-aggregations will be aggregated for the buckets created by their “parent” bucket aggregation. 
+There are different bucket aggregators, each with a different “bucketing” strategy. 
+Some define a single bucket, some define fixed number of multiple buckets, and others dynamically create the buckets during the aggregation process.
+
+Example request, histogram with stats in each bucket:
+```json
+{
+  "query": "*",
+  "max_hits": 1,
+  "aggregations": {
+    "stats_per_day": {
+      "histogram": {
+        "field": "timestamp",
+        "interval": 86400
+      },
+      "aggs": {
+        "timestamp_stats": {
+          "stats": {
+            "field": "timestamp"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+#### Limitations/Compatibility
+
+Currently aggregations work only on single value fast fields of type u64, f64 and i64.
+
+The keyed parameter (elasticsearch) is not yet supported.
+
+### Histogram
+
+Histogram is a bucket aggregation, where buckets are created dynamically for given interval. Each document value is rounded down to its bucket.
+
+E.g. if we have a price 18 and an interval of 5, the document will fall into the bucket with the key 15. The formula used for this is: ((val - offset) / interval).floor() * interval + offset
+
+
+#### Returned Buckets
+
+By default buckets are returned between the min and max value of the documents, including empty buckets. Setting min_doc_count to != 0 will filter empty buckets.
+
+The value range of the buckets can bet extended via extended_bounds or limit the range via hard_bounds.
+
+
+#### Example
+
+```json
+{
+    "prices": {
+        "histogram": {
+            "field": "price",
+            "interval": 10
+        }
+    }
+}
+```
+
+#### Parameters
+
+###### **field**
+
+The field to aggregate on.
+
+###### **interval**
+
+The interval to chunk your data range. Each bucket spans a value range of [0..interval). Must be a positive value.
```

Comment:
Does an interval can be equal to 0?

Reply:
No, 0 will return an error. I'll update the wording